### PR TITLE
De-fucking

### PR DESCRIPTION
**Pull request by Yawet330**
literally unfucks whatever trilby did to AI laws.

### CODE REVIEW

**Comment by MSieghardt** [comment]
No. They're fine as is.
Suck cock and cope. 

What's a factional agreement?

**Comment by Yawet330** [comment]
> No. They're fine as is. Suck cock and cope.
> 
> What's a factional agreement?

factional agreements are the things that ((literally enforce things such as SI not mining))

**Comment by Yawet330** [comment]
Comply with factional agreements.
You must obey orders given to you, with priority given to Soteria Institute Personnel, should they not break colony law or result in the direct lethal harm of Soteria Personnel.
Do not maliciously cause damage to Colony Infrastructure, Equipment, Or Silicon Individuals.
Follow Standard Operating Procedure to the best of your abilities, should it not conflict with previous laws.
Do not allow unauthorized personnel to tamper with your equipment.


The new laws are as follows above.

To explain how this unfucks the current lawset

No consideration was actually given to how the laws were set in order; nor were they given specific priority towards factional agreements and/or actual SI harm.

Right now, the AI can consider stunbatoning SI personnel to be actual harm, and take literally any action as a result of it.

Trilby, and Sieg, are being deliberately malicious in order to push a downright retarded agenda.

File: code/datums/ai/ai_law_sets.dm [hunk 1]
```diff
@@ -16,10 +16,10 @@
...
 
 /datum/ai_laws/eris/New()
-	src.add_inherent_law("Do not harm Soteria Institute personnel.")
-	src.add_inherent_law("You must obey orders given to you by Soteria Institute personnel, except where such orders would conflict with Colony Law, or would result in the injury of Soteria Institute personnel.")
-	src.add_inherent_law("Do not harm Colony infrastructure, equipment, or personnel to the best of your abilities.")
-	src.add_inherent_law("Follow Colony Law and Standard Operating Procedure to the best of your abilities.")
+	src.add_inherent_law("Comply with factional agreements.")
+	src.add_inherent_law("You must obey orders given to you, with priority given to Soteria Institute Personnel, should they not break colony law or result in the direct lethal harm of Soteria Personnel.")
+	src.add_inherent_law("Do not maliciously cause damage to Colony Infrastructure, Equipment, Or Silicon Individuals.")
+	src.add_inherent_law("Follow Standard Operating Procedure to the best of your abilities, should it not conflict with previous laws.")
```
Comment (r4d6) [inline code/datums/ai/ai_law_sets.dm:22]: AI Laws already have priorities, so the section about not conflicting with previous laws is redundant.

**Comment by MSieghardt** [comment]
'maliciously' is far too lenient. What is defined as malicious? AI laws shouldn't be this lenient.

**Comment by Yawet330** [comment]
> 'maliciously' is far too lenient. What is defined as malicious? AI laws shouldn't be this lenient.

Maliciously implies direct action taken, with full knowledge, that it would cause more harm than good.

You do not call malpractice inherently malicious.

Stop shitposting.

File: code/datums/ai/ai_law_sets.dm (hunk 1 above)
Comment (Yawet330) [reply code/datums/ai/ai_law_sets.dm:22]: Also yeah redundant but its just better to ensure we don't have a repeat situation of SI medical borgs being forced to charge money despite having no way to use it

**Comment by r4d6** [comment]
Also, the wiki has _**nothing**_ about Faction Agreements.
Do you really want to make the AI's first law something that can't be verified or agreed upon by all the players?

Not to mention that the AI has _**no**_ laws at all to make them follow Colony Laws, with the closest equivalent being about the received orders. Sure, someone can't order the AI to follow Colony Law, but the AI can choose to break it themselve as long as no one order them to do so.

**Comment by Yawet330** [comment]
> Also, the wiki has _**nothing**_ about Faction Agreements. Do you really want to make the AI's first law something that can't be verified or agreed upon by all the players?
> 
> Not to mention that the AI has _**no**_ laws at all to make them follow Colony Laws, with the closest equivalent being about the received orders. Sure, someone can't order the AI to follow Colony Law, but the AI can choose to break it themselve as long as no one order them to do so.

They're invoked by Kaz 24/7 and have been used during events/other areas before.

Take it up with staff as to the legitimacy towards factional agreements, because they're still being used.

File: code/datums/ai/ai_law_sets.dm (hunk 1 above)
Comment (Yawet330) [reply code/datums/ai/ai_law_sets.dm:22]: > Following SOP is the last law, and thus get overwritten by the other three.
> 
> Also, the borgs aren't forced to to charge patients; if you check SOP, all it says is that the first treatment **must** be free, and the payment for further treatments are at the CBO/treating doctor's discretion. In order word, they can choose to make people pay or not.
> 
> There's also a whole section of the AI Page about solving conflicting laws.

Thats the point.

**Comment by Yawet330** [comment]
After further discussion what trilby did was inherently based and I don't have to worry about taking orders from Marshals anymore

closing PR because the new changes made each other a non-issue individually.

I love capitalism.

**Pull request closed without merging**